src/App.js

```diff
@@ -1,43 +1,87 @@
 import React from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import PropTypes from "prop-types";
 
 class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
+  state = {
+    jokeData: {
+      results: []
+    },
+    searchTerm: "",
+    loadingState: "resting"
+  };
+
+  handleChange = event => {
+    this.setState({ searchTerm: event.target.value });
+  };
+
+  handleClick = event => {
+    event.preventDefault();
+    this.setState({ loadingState: "loading" });
+  };
+
+  async fetchJokes() {
+    try {
+      const url =
+        this.state.searchTerm === ""
+          ? `https://icanhazdadjoke.com/search`
+          : `https://icanhazdadjoke.com/search?term=${this.state.searchTerm}`;
+      const fetchUrl = fetch(url, {
+        method: "GET",
+        headers: { accept: "application/json" }
+      });
+      const response = await fetchUrl;
+      const myData = await response.json();
+      this.setState({
+        jokeData: { results: myData.results },
+        loadingState: "loaded"
+      });
+    } catch (e) {
+      this.setState({ loadingState: "Something went wrong" });
     }
   }
 
-  async componentDidMount() {
-    const url = 'https://icanhazdadjoke.com/search?term="cat"';
-    const fetchUrl = fetch(url, {
-      method: "GET",
-      headers: { accept: "application/json" }
-    });
-    const response = await fetchUrl;
-    const myData = await response.json();
-    this.setState({ jokeData: { results: myData.results } });
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.loadingState === "loading") {
+      this.fetchJokes();
+    }
+  }
+
+  componentDidMount() {
+    this.fetchJokes();
   }
 
   // This needs rendering inside a map function, then rendering in the render as <div>{jokeList}</div>
   // <Joke key="[ENTER THE KEY HERE]" joke={this.state}/>
   render() {
     const jokeList = this.state.jokeData.results.map(item => (
```

Comment:
I know why you did it this way, but this causes this variable to created and for the iterating inside `map` to happen even when `loadingState` isn't "loaded". Usually these kinds of things are "in-lined" down on line 75.